File: src/lib/scala/Slate.Common/src/main/scala/slate/common/validations/ValidationResults.scala
/**
  * <slate_header>
  * author: Kishore Reddy
  * url: https://github.com/kishorereddy/scala-slate
  * copyright: 2016 Kishore Reddy
  * license: https://github.com/kishorereddy/scala-slate/blob/master/LICENSE.md
  * desc: a scala micro-framework
  * usage: Please refer to license on github for more info.
  * </slate_header>
  */
package slate.common.validations

import slate.common.Reference

case class ValidationResults(success : Boolean                 ,
                             msg     : String            = ""  ,
                             ref     : Option[Reference] = None,
                             code    : Int               = 0   ,
                             results : Option[List[ValidationResult]])
{
  // Any errors ?
  val hasErrors = results.fold(false)( e => e.isEmpty )
}


object ValidationResults {

  def apply(errors:Option[List[ValidationResult]]): ValidationResults = {
    val success = errors.isEmpty || errors.fold(true)( e => e.isEmpty )
    val message = errors.fold("")( e => e.head.msg.getOrElse("") )
    val code = if (success) 1 else 0
    new ValidationResults(success, message, None, code, errors)
  }
}

File: src/lib/kotlin/slatekit-common/src/main/kotlin/slatekit/common/validations/ValidationResults.kt
/**
 * <slate_header>
 * url: www.slatekit.com
 * git: www.github.com/code-helix/slatekit
 * org: www.codehelix.co
 * author: Kishore Reddy
 * copyright: 2016 CodeHelix Solutions Inc.
 * license: refer to website and/or github
 * about: A tool-kit, utility library and server-backend
 * mantra: Simplicity above all else
 * </slate_header>
 */

package slatekit.common.validations

data class ValidationResults(val success: Boolean,
                             val msg: String = "",
                             val ref: Reference? = null,
                             val code: Int = 0,
                             val results: List<ValidationResult>?) {
    // Any errors ?
    val hasErrors = results?.isEmpty() ?: false


    companion object {

        @JvmStatic
        fun build(errors: List<ValidationResult>?): ValidationResults {
            val success = errors?.isEmpty() ?: true
            val message = errors?.first()?.msg ?: ""
            val code = if (success) 1 else 0
            return ValidationResults(success, message, null, code, errors)
        }
    }
}

